week 6.cpp

```diff
@@ -0,0 +1,204 @@
+#include <iostream>
+#include <stdlib.h>
+using namespace std;
+int id=0;
+class Item
+{
+	int item_id;
+	//char item_name;
+	float price;
+	
+	
+	public:
+
+	int item_count;
+	Item()
+	{
+		item_id=id++;
+		price=10;
+		item_count=2;
+		
+	}
+	float get_price()
+	{
+	   return price;
+	}
+
+	void decrease_item_count()
+	{
+
+		item_count--;
+
+	}
+
+	void increase_item_count()
+	{
+
+		item_count++;
+
+	}
+
+
+
+};
+
+class coin
+{
+  int count_1; // no of 1 rupee coins
+  int count_5;  // no of 5 rupee coins
+  int count_10; // no of 10 rupee coins
+  
+  public:
+  coin()
+  {
+  	count_1=20;
+  	count_5=10;
+  	count_10=3;
+  }
+	
+};
+
+class Vending_Machine
+{
+	float current_balance;  // current_balance = count_1 + count_5 + count_10
+	int available_items[3];
+	coin* coin_obj; 
+
+	public:
+
+	Item* items_list[3];
+	Vending_Machine()
+	{
+		current_balance=100; 
+		for (int i=0;i<3;i++)
+		{ items_list[i]= new Item();
+		  available_items[i]=1; 
+		   	
+		}
+		
+		coin_obj=new coin();	
+	}
+	
+    
+    void pay_change(float num)
+    {
+      float sum=0;
+      
+      if(current_balance>=num)
+      { 
+	      while(sum<=num && count_10>0)
+	      {
+	      	sum+=10;
+	      	count_10-=1;
+
+	      }
+	      while(sum<=num && count_5>0)
+	      {
+	      	sum+=5;
+	      	count_5-=1;
+	      	
+	      }
+
+	      while(sum<=num && count_1>0)
+	      {
+	      	sum+=1;
+	      	count_1-=1;
+	      	
+	      }
```

Comment:
This while loop code looks awfully similar, between the three coin types.  It might be worth refactoring into a common method